src/android/shakeDetector.java

```diff
@@ -0,0 +1,251 @@
+                } catch (Exception e) {
+                    Log.e(context, TAG, "Got error "+e.getMessage()+" while processing object "
+                            + notifyConfigs.getJSONObject(i) + " at index "+i);
+                }
+            }
+        } catch(JSONException e) {
+            Log.e(context, TAG, e.getMessage());
+            Log.e(context, TAG, e.toString());
+        }
+    }
+
+    private void mergeObjects(JSONObject existing, JSONObject autogen) throws JSONException {
+        JSONArray toBeCopiedKeys = autogen.names();
+        for(int j = 0; j < toBeCopiedKeys.length(); j++) {
+            String currKey = toBeCopiedKeys.getString(j);
+            existing.put(currKey, autogen.get(currKey));
+        }
+    }
+
+    private int findEntryWithId(JSONArray array, long id) throws JSONException {
+        if (array == null) {
+            return -1;
+        }
+        for (int i = 0; i < array.length(); i++) {
+            JSONObject currCheckedObject = array.getJSONObject(i);
+            if (currCheckedObject.getLong(ID) == id) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private void storeDataAndNotify(Context context, float[] accelerations, long currTime, float combinedAccel) throws JSONException{
+
+        JSONObject incidentData = new JSONObject();
+        UserCache uc = UserCacheFactory.getUserCache(this);
+        SimpleLocation loc[] = uc.getLastSensorData(R.string.key_usercache_filtered_location, 1, SimpleLocation.class);
+
+        if (loc.length == 0) {
+            incidentData.put("inc_ts", ((double) currTime) / 1000);
+        } else {
+            SimpleLocation mostRecentLocation = loc[0];
+            incidentData.put("inc_ts", ((double) currTime) / 1000);
+            incidentData.put("inc_latitude", mostRecentLocation.getLatitude());
+            incidentData.put("inc_longitude", mostRecentLocation.getLongitude());
+            incidentData.put("inc_altitude", mostRecentLocation.getAltitude());
+            incidentData.put("inc_speed", mostRecentLocation.getSensed_speed());
+            incidentData.put("inc_accuracy", mostRecentLocation.getAccuracy());
+            incidentData.put("inc_bearing", mostRecentLocation.getBearing());
+        }
+
+        //int modeOfTransportation = -1;
+
+        //Storage of mode of transportation (not implemented on iOS)
+        /*
+        ActivityRecognitionResult currActivity[] = uc.getLastSensorData(R.string.key_usercache_activity, 1, ActivityRecognitionResult.class);
+
+        if ((currActivity.length != 0) && (currActivity[0] != null)){
+            DetectedActivity mostProbableActivity = currActivity[0].getMostProbableActivity();
+            modeOfTransportation = mostProbableActivity.getType();
+        }
+
+        incidentData.put("inc_mode", modeOfTransportation);
+        */
+
+        incidentData.put("inc_xAccel", accelerations[0]);
+        incidentData.put("inc_yAccel", accelerations[1]);
```

Comment:
Again, there should be a wrapper for the acceleration, and it should be stored as an object directly. That will make this much more modular.